src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const types = [];
+  const store = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties': Object.assign(store, action.extraData);
+        break;
+      case 'removeProperties':
+        if (action.keysToRemove.length > 0) {
+          for (const key of action.keysToRemove) {
+            delete store[key];
+          }
+        }
```

Comment:
you don't need this condition
if the array is empty itself, iteration won't start

Reply:
done